recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,94 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class WasmtimeConan(ConanFile):
+    name = 'wasmtime'
+    homepage = 'https://github.com/bytecodealliance/wasmtime'
+    license = 'Apache-2.0'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "Standalone JIT-style runtime for WebAssembly, using Cranelift"
+    topics = ("webassembly", "wasm", "wasi")
+    settings = "os", "compiler", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "apple-clang": "9.4",
+            "clang": "3.3",
+            "gcc": "4.9.4"
+        }
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        compiler = self.settings.compiler
+        min_version = self._minimum_compilers_version[str(compiler)]
+        if tools.Version(compiler.version) < min_version:
+            msg = (
+                "{} requires C{} features which are not supported by compiler {} {} !!"
+            ).format(self.name, self._minimum_cpp_standard, compiler, compiler.version)
+            raise ConanInvalidConfiguration(msg)
```

Comment:
We like to print a warning since user have custom settings.... check the template in conan-io/conan#8002

Reply:
@uilianries 
I will add this change and it will broke my receipt

Reply:
@prince-chrismc Warning is printing in except:
```python
            msg = (
                "{} recipe lacks information about the {} compiler, "
                "support for the required C{} features is assumed"
            ).format(self.name, compiler, self._minimum_cpp_standard)
            self.output.warn(msg)
```